File: cpp/Program/Garage.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"
#include <vector>

using namespace std;

class Garage
{
private:
    string namaGarasi;
    int luasGarasi;
    std::vector<Vehicle> vehicles;

public:
    Garage(){

    }

    Garage(string namaGarasi){
        this->namaGarasi = namaGarasi;
        luasGarasi = 0;
        this->vehicles = {};
    }

    Garage(string namaGarasi, int luasGarasi){
        this->namaGarasi = namaGarasi;
        this->luasGarasi = luasGarasi;
        this->vehicles = {};
    }

    Garage(string namaGarasi, int luasGarasi, std::vector<Vehicle> vehicles){
        this->namaGarasi = namaGarasi;
        this->luasGarasi = luasGarasi;
        vehicles = vehicles;
    }


    string GetNamaGarasi()  {
        return namaGarasi;
    }

    void SetNamaGarasi(string namaGarasi) {
        this->namaGarasi = namaGarasi;
    }

    int GetLuasGarasi()  {
        return luasGarasi;
    }

    void SetLuasGarasi(int luasGarasi) {
        this->luasGarasi = luasGarasi;
    }

    std::vector<Vehicle> GetVehicles()  {
        return vehicles;
    }

    void SetVehicles(std::vector<Vehicle> vehicles) {
        vehicles = vehicles;
    }

    void addVehicle(Vehicle v) {
        vehicles.push_back(v);
    }

    ~Garage(){

    }
};

File: python/Program/Garage.py
from Vehicle import Vehicle

class Garage:
    __namaGarasi = ""
    __luasGarasi = ''
    
    
    def __init__(self, namaGarasi="", luasGarasi=0):
        self.__namaGarasi = namaGarasi
        self.__luasGarasi = luasGarasi
        self.__kendaraan = []


    def get_namaGarasi(self):
        return self.__namaGarasi

    def set_namaGarasi(self, namaGarasi):
        self.__namaGarasi = namaGarasi

    def get_luasGarasi(self):
        return self.__luasGarasi

    def set_luasGarasi(self, luasGarasi):
        self.__luasGarasi = luasGarasi

    def get_Kendaraan(self):
        return self.__kendaraan

    def set_Kendaraan(self, value):
        if not self.__kendaraan:  # Jika daftar kendaraan kosong
            self.__kendaraan.append(Vehicle(platNomor = "", merk= "", tahunProduksi= "", warna= ""))  # Tambahkan kendaraan kosong
        self.__kendaraan.append(value)  # Tambahkan kendaraan ke daftar

